core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java

```diff
@@ -471,14 +493,18 @@ private void showTabSwitcher() {
     progressBar.setVisibility(View.GONE);
     backToTopButton.hide();
     tabSwitcherRoot.setVisibility(View.VISIBLE);
-    tabSwitcherRoot.startAnimation(AnimationUtils.loadAnimation(this, R.anim.slide_down));
+    setAnimation(tabSwitcherRoot, this, R.anim.slide_down);
     if (tabsAdapter.getSelected() < webViewList.size() &&
       tabRecyclerView.getLayoutManager() != null) {
       tabRecyclerView.getLayoutManager().scrollToPosition(tabsAdapter.getSelected());
     }
     mainMenu.showTabSwitcherOptions();
   }
 
+  private void setAnimation(View view, Context context, int anim) {
```

Comment:
`setAnimation` as a name could be improved. Avoid using `setXXX` unless you are actually setting a field of an object.
here what we are doing is starting an animation so I am okay with `startAnimation`

We don't need to pass in the context here as we already pass in a view. Replace `AnimationUtils.loadAnimation(context, anim)` with `AnimationUtils.loadAnimation(view.getContext(), anim)`. After you do that the `context` parameter should be greyed out, hit `alt+enter` and you will get the option to delete it and remove it from the callsites

One of the annoying bits of android is passing around these id ints that have no type safety, to try and help this situation androidx has annotations that help enforce some safety around their usage. annotate `int anim` with `@AnimRes`

This is mostly there, in an ideal world we could write this as a kotlin extension function on view and it would look like
```
fun View.startAnimation(@AnimRes animId:Int){
  startAnimation(AnimationUtils.loadAnimation(context, animId))
}
```
but seeing as how the callsites are all in java it would end up looking like `ViewExtensionsKt.startAnimation(view, R.anim.my_anim)` anyway so that is not particularly relevant as of right now

Reply:
Thanks a lot for the pointers @macgills :)
I'll make the changes.

Reply:
@macgills is it fine now?

Reply:
Just one more change